backend/core/src/auth/services/user.service.ts

```diff
@@ -621,7 +621,16 @@ export class UserService {
   }
 
   private static getPublicConfirmationUrl(appUrl: string, user: User) {
-    return `${appUrl}?token=${user.confirmationToken}`
+    const urlObj = new URL(appUrl)
+
+    const redirectUrl = urlObj.searchParams.get("redirectUrl")
+    const listingId = urlObj.searchParams.get("listingId")
+
+    const confirmationUrl =
```

Comment:
Nit: A suggestion here to reduce repetition / something slightly more readable might just be an if statement that pushes the redirect onto the string (same as below).